src/entity/Entity.php

```diff
@@ -665,8 +665,7 @@ protected function entityBaseTick(int $tickDiff = 1) : bool{
 				$block = $entityBlock->getSide($face);
 				foreach ($block->getCollisionBoxes() as $blockBox) {
 					if ($entityBox->intersectsWith($blockBox)) {
-						$block->onEntityCollide($this, Facing::opposite($face));
-						$hasUpdate = true;
+						$hasUpdate = $block->onEntityCollide($this, Facing::opposite($face));
```

Comment:
Don't do this. If the block isn't updated, it'll trash any previous value for $hasUpdate. (This is also why I questioned doing this in entityBaseTick...)